lib/CXGN/BrAPI/v2/Observations.pm

```diff
@@ -55,22 +56,27 @@ sub search {
     my $start_index = $page*$page_size;
     my $end_index = $page*$page_size + $page_size - 1;
 
-    my ($data, $unique_traits)  = _search_observation_id(
-            $self->bcs_schema,
-            $observation_level,
-            $trial_ids,
-            $trial_ids_arrayref,
-            1,
-            $season_arrayref,
-            $location_ids_arrayref,
-            $accession_ids_arrayref,
-            $program_ids_arrayref,
-            \@observation_variable_db_ids,
-            $observation_db_id,
-            $observation_unit_db_id, #plot_list
-            $limit,
-            $offset,
+    my $phenotypes_search = CXGN::Phenotypes::SearchFactory->instantiate(
+        'MaterializedViewTable',
+        {
+            bcs_schema=>$self->bcs_schema,
+            data_level=>$observation_level,
+            trial_list=>$trial_ids,
+            folder_list=>$trial_ids_arrayref,
+            include_timestamp=>1,
+            year_list=>$season_arrayref,
+            location_list=>$location_ids_arrayref,
+            accession_list=>$accession_ids_arrayref,
+            program_list=>$program_ids_arrayref,
+            trait_list=>\@observation_variable_db_ids,
+            trait_contains=>\@observation_variable_names,
+            plot_list=>$observation_unit_db_id,
+            limit=>$limit,
+            offset=>$offset,
+            order_by=>"plot_number"
+        }
     );
+    my ($data, $unique_traits) = $phenotypes_search->search();
```

Comment:
Refactored the `_search_observation_id` method in the `Observations.pm` class to be in the `CXGN/Phenotypes/Search/MaterializedViewTable.pm#search` function